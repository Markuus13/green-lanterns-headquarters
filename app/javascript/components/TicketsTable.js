import React from 'react'
import PropTypes from 'prop-types'

const TicketsTable = ({ tickets }) => (
  <table>
    <thead>
      <tr>
        <th>Número do Chamado</th>
        <th>Título</th>
        <th>Descrição</th>
      </tr>
    </thead>
    <tbody>
      {tickets.map((ticket) => (
        <tr key={ticket.order_number}>
          <td>{ticket.order_number}</td>
          <td>{ticket.title}</td>
          <td>{ticket.description}</td>
        </tr>
      ))}
    </tbody>
  </table>
)

TicketsTable.propTypes = {
  tickets: PropTypes.arrayOf(
    PropTypes.shape({
      order_number: PropTypes.string,
      title: PropTypes.string,
      description: PropTypes.string
    })
  )
}

export default TicketsTable