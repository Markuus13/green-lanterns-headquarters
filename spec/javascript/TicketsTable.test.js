import React from 'react';
import TicketsTable from 'components/TicketsTable';
import renderer from 'react-test-renderer';

const mockTickets = () => (
  [
    {
      order_number: '1',
      title: 'first',
      description: 'ameno'
    },
    {
      order_number: '2',
      title: 'second',
      description: 'dorime'
    }
  ]
)

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <TicketsTable
      tickets={mockTickets()}
    />
  );
  expect(component.toJSON()).toMatchSnapshot();
});