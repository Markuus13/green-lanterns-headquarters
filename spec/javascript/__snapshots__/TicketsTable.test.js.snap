// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<table>
  <thead>
    <tr>
      <th>
        Número do Chamado
      </th>
      <th>
        Título
      </th>
      <th>
        Descrição
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        1
      </td>
      <td>
        first
      </td>
      <td>
        ameno
      </td>
    </tr>
    <tr>
      <td>
        2
      </td>
      <td>
        second
      </td>
      <td>
        dorime
      </td>
    </tr>
  </tbody>
</table>
`;
